### PR TITLE
[ACR] Usability improvements, add '--no-wait' to 'acr build'

### PR DESCRIPTION
**Pull request by djyou**
Fixes https://github.com/Azure/azure-cli/issues/8798
Fixes https://github.com/AzureCR/azure-cli/issues/306

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by djyou** [comment]
```
>az acr repository list -n myregistry
Looks like you don't have access to registry 'myregistry.azurecr.io'. Are firewalls and virtual networks enabled?
```
/cc @sajayantony @yuwaMSFT2

File: src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/_docker_utils.py [hunk 1]
```diff
@@ -176,9 +176,12 @@ def _get_credentials(cmd,  # pylint: disable=too-many-statements
             registry_name, '-{}'.format(tenant_suffix) if tenant_suffix else '', login_server_suffix).lower()
 
     # Validate the login server is reachable
-    challenge = 'https://' + login_server + '/v2/'
+    url = 'https://' + login_server + '/v2/'
     try:
-        requests.get(challenge, verify=(not should_disable_connection_verify()))
+        challenge = requests.get(url, verify=(not should_disable_connection_verify()))
```
Comment (djyou) [inline src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/_docker_utils.py]: @yuwaMSFT2 All runtime requests (including classic) will first run through this check to make sure it's reachable and accessible.

**Comment by sajayantony** [comment]
LGTM 👍

**Pull request merged**